app/scripts/modules/appengine/pipeline/stages/enableAsg/appengineEnableAsgStage.ts

```diff
@@ -0,0 +1,49 @@
+import {module} from 'angular';
+
+import {IAppengineStageScope} from 'appengine/domain/index';
+import {ACCOUNT_SERVICE, AccountService} from 'core/account/account.service';
+import {AppengineStageCtrl} from '../appengineStage.controller';
+
+class AppengineEnableAsgStageCtrl extends AppengineStageCtrl {
+  static get $inject() { return ['$scope', 'accountService']; }
+
+  constructor(public $scope: IAppengineStageScope, accountService: AccountService) {
```

Comment:
For posterity, the presence of access modifiers tells the compiler to create a local reference of the ctor arg.  As expected, the complied code makes no distinction with the access modifiers, thus the modifiers are purely for compile-time checking.  Omitting the access modifier will _NOT_ create a class-level field reference.

In NG2, the access modifier of class properties referenced in component templates must be non-`private` or the compiler will complain as the component templates are essentially turned into TS classes.  This is not an issue in NG1, but for easier migration, any class-level fields referenced in templates should not be `private`.

In general, the most restrictive access for fields, methods, classes should be chosen - even though this same level of security will not carry over to the transpiled code.